protocol/v1/qbft/instance/commit.go

```diff
@@ -17,8 +17,26 @@ import (
 
 // CommitMsgPipeline - the main commit msg pipeline
 func (i *Instance) CommitMsgPipeline() pipelines.SignedMessagePipeline {
+	validationPipeline := i.CommitMsgValidationPipeline()
 	return pipelines.Combine(
-		i.CommitMsgValidationPipeline(),
+		signedmsg.CheckProposal(i.State()),
```

Comment:
@nkryuchkov it's redundant as we check it again in `protocol/v1/qbft/instance/commit.go:39`

Reply:
@nivBlox Yes, the proposal is checked twice right now
1) here, because spec does this as the first check upon commit message
2) in `ValidateProposal` (`protocol/v1/qbft/instance/commit.go:39`) to avoid panic if `ValidateProposal` is called when `ProposalAcceptedForCurrentRound` is `nil` (it's called only when `ProposalAcceptedForCurrentRound` is not `nil`, however, if `ValidateProposal` is reused or called in tests, this validation might be needed)

So if we remove the check in 1) (here), the order of checks will be different than in spec, so if we remove the redundancy, I would prefer to do that in 2). Would you agree with that?

Reply:
As disscused, removing `signedmsg.CheckProposal(i.State()),` as we check it again on other func

Reply:
@nivBlox I've checked if removing `signedmsg.CheckProposal(i.State())` breaks something and it does: the `no previous accepted proposal` test expects the `did not receive proposal for this round` error, but if we remove the check here, the error will be wrapped: `invalid commit message: did not receive proposal for this round`, so the test will fail.